### PR TITLE
Add back maven cache

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**
- check if the problem was fixed in the meantime

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -262,6 +262,15 @@ stages:
           artifact: RuleDescriptorBin
           targetPath: analyzers\src\SonarAnalyzer.RuleDescriptorGenerator\bin
 
+      - task: Cache@2
+        displayName: "Cache Maven local repo"
+        inputs:
+          key: 'maven | "$(Agent.OS)" | **/pom.xml'
+          restoreKeys: |
+            maven | "$(Agent.OS)"
+            maven
```
Comment (pavel-mikula-sonarsource) [inline azure-pipelines.yml:271]: This should not be here, as we don't need it (we never populate it with something useful).
```suggestion
```

File: azure-pipelines.yml [hunk 2]
```diff
@@ -262,6 +262,15 @@ stages:
           artifact: RuleDescriptorBin
           targetPath: analyzers\src\SonarAnalyzer.RuleDescriptorGenerator\bin
 
+      - task: Cache@2
+        displayName: "Cache Maven local repo"
+        inputs:
+          key: 'maven | "$(Agent.OS)" | **/pom.xml'
```
Comment (pavel-mikula-sonarsource) [inline azure-pipelines.yml:268]: We should include the version in the key to avoid infinite grow. With help of the script you refactored recently.

**Review by pavel-mikula-sonarsource — CHANGES_REQUESTED** [review]
Let's add the improvements we prepared in the investigation.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -262,6 +262,14 @@ stages:
           artifact: RuleDescriptorBin
           targetPath: analyzers\src\SonarAnalyzer.RuleDescriptorGenerator\bin
 
+      - powershell: .\scripts\build\set-azp-project-version-variable.ps1
+
+      - task: Cache@1
```
Comment (pavel-mikula-sonarsource) [inline azure-pipelines.yml:267]: Is it version 1 on purpose? Did you try to play with v2? I don't know what the difference is.

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM, there's just a question about version intention. :shipit:

File: azure-pipelines.yml (hunk 3 above)
Comment (costin-zaharia-sonarsource) [reply azure-pipelines.yml:267]: Yes, I've tried again with version 2 and it failed.

**Review by pavel-mikula-sonarsource — COMMENTED** [review]
~~LGTM~~ (_edit:_ Bad PR)

**Pull request merged**